src/grid/atomgrid.py

```diff
@@ -288,6 +291,108 @@ def convert_cart_to_sph(self, points=None, center=None):
         center = self.center if center is None else np.asarray(center)
         return convert_cart_to_sph(points, center)
 
+    def fit_values(self, value_array):
+        """Fit given value arrays into splines that matches atomic grid.
+
+        Parameters
+        ----------
+        value_array : np.ndarray(N,)
+            a one dimentional array evaluated at each radial grid point
+
+        Returns
+        -------
+        list[scipy.PPoly]
+            A list of cubic spline fitted by given value arrays
+        """
+        if self._basis is None:
+            theta, phi = self.convert_cart_to_sph().T[1:]
+            self._basis = self._generate_real_sph_harm(self.l_max // 2, theta, phi)
+        print(self._basis.shape)
+        prod_value = self._basis * value_array * self.weights
+        rad_values = [
+            np.sum(prod_value[:, self.indices[i] : self.indices[i + 1]], axis=-1)
+            for i in range(self.n_shells)
+        ]
+        # rad_values in shape (n_shell, n_sph_harms)
+
+        ml_sph_values = np.array(rad_values).T  # shape(n_sph_harms, n_shell)
+        ml_sph_values /= self.rgrid.points ** 2 * self.rgrid.weights
+        return [
+            CubicSpline(x=self.rgrid.points, y=sph_val) for sph_val in ml_sph_values
+        ]
+
+    def interpolate(self, points, splines, deriv=0):
```

Comment:
Following `scipy.interpolate` API and the cubic grid's `interpolate` method in PR #133, this should take the `points` and `values` (not `splines`). This would be more user-friendly as well. Can you please fix it?

Reply:
So to combine the fitting and interpolate functions together?

Reply:
Well, `interpolate` uses the fitting function under the hood, but we should keep the fitting function. Does that make sense?

Reply:
It makes sense and `fitting` function need to be kept for other functionality. But in this way, everytime the `interpolate` being called, it need to recompute the splines. If that's not a very frequently happened case, I'm fine with the change.

Reply:
That shouldn't be a problem, as I cannot see the case that it would be called repeatedly, also the `interpolate` method takes an array of points. Passing `points` & `values` makes it easier to use and more consistent (with Scipy and CUbic Grid which cannot return a spline all the time). So, I think it's best to go with it for now.

Reply:
OK. Will change the code accordingly

Reply:
I think it's OK. It's a lot more user friendly to fit to values. Since we support an array of points, I doubt it will often be that we repeatedly need to repeat things. If we preserve low-level functionality with Derrick's previous strategy, then an (expert) user could use that method if (s)he needed to avoid repeated evaluation of splines, while at the (general) user level the intuitive points/values syntax was supported.  Does this make sense @FarnazH @tczorro

Reply:
That makes sense. I have updated the API and put some extra notes/comments. Let me know if any extra modification is needed